### ISSUE TITLE
 An unknown server-side error occurred while processing the command . Original error: Cannot read properties of undefined (reading 'match') when using appium flutter finder

### ISSUE THREAD

**Issue by lamdev99** [appium-userland/appium-flutter-driver#313]
When I tried to implement example app in [example.py](https://github.com/appium-userland/appium-flutter-driver/blob/master/example/python/example.py), which is in your project's example, I have met this issue

selenium.common.exceptions.WebDriverException: Message: An unknown server-side error occurred while processing the command. Original error: Cannot read properties of undefined (reading 'match')
Stacktrace:
UnknownError: An unknown server-side error occurred while processing the command. Original error: Cannot read properties of undefined (reading 'match')

Full of my code: 

import os
from time import sleep

from appium.webdriver import Remote
from appium_flutter_finder.flutter_finder import FlutterElement, FlutterFinder

driver = Remote('http://localhost:4723/wd/hub', dict(
    platformName='Android',
    automationName='flutter',
    platformVersion='8',
    deviceName='emulator-5556',
    app='${PATH}/app-debug.apk'.format(
      os.path.dirname(os.path.realpath(__file__)))
))

finder = FlutterFinder()

tooltip_finder = finder.by_value_key("increment")
floating_button_element = FlutterElement(driver, tooltip_finder)
floating_button_element.click()

**Looking forward to hearing from you soon**

**Comment by KazuCocoa** [comment]
Could you share the appium server log as GIST?
Btw, Appium 2 and the latest appium-flutter-driver may help.

**Comment by lamdev99** [comment]
Hi, here is my [server log ](https://gist.github.com/lamdev99/7b7ab103e2e2ae5068cf476882948bd4)
I have already updated to appium 2 and lastest appium-flutter-driver

**Comment by dorbluevine** [comment]
Also experiencing a similar issue since updating to flutter 3 (@lamdev99 have you updated your app to flutter 3?)

Our infrastructure is in JS (webdriver.io), when we are running the app is installed on the device, opened and then immediately crashes (it does not crash when opened manually).

The error we are getting is this:
```
DEBUG webdriver: request failed due to response error: unknown error
ERROR webdriver: Request failed with status 500 due to unknown error: An unknown server-side error occurred while processing the command. Original error: Cannot read property 'match' of undefined
ERROR webdriver: unknown error: An unknown server-side error occurred while processing the command. Original error: Cannot read property 'match' of undefined
```

Downgrading to flutter 2.10 solves this.

We tried setting up a new minimal flutter project and a new minimal webdriver.io appium-flutter-driver based project, and it behaved the same - works with flutter 2, crashes with flutter 3.

We are trying to understand this issue since last week and still have no idea what causes it.

We are using appium-flutter-driver version 0.0.35, any chance flutter 3 broke compatibility?

**Comment by KazuCocoa** [comment]
Could you share the full log to understand the environment?

Afaik, https://github.com/appium-userland/appium-flutter-driver/pull/283 helps to fix flutter 3 change, that is available in Appium 2 and the latest this project.
If still happens, could you share the logcat or systemlog as GIST? Maybe this driver failed to find a websocket URL in the log

**Comment by lamdev99** [comment]
@dorbluevine I have downgraded to Flutter 2.10, and it ran, wonderfull!!!

**Comment by lamdev99** [comment]
@KazuCocoa Here is my full server log again [server log](https://gist.github.com/lamdev99/7b7ab103e2e2ae5068cf476882948bd4). 
Thanks

**Comment by KazuCocoa** [comment]
Thanks, but the log used `Appium v1.22.3 creating new FlutterDriver (v0.0.32) session`.

> I have already updated to appium 2 and lastest appium-flutter-driver

Do you have this one?

**Comment by lamdev99** [comment]
@KazuCocoa  I have re-checked flutter version again with your current update 1.5.0 and it worked. 
Many thanks. @KazuCocoa

**Comment by lamdev99** [comment]
@dorbluevine  let try to update your appium flutter driver, it have some big updates and maybe can solve your issus with Flutter 3 xD

**Comment by dorbluevine** [comment]
**UPDATE:**
It seems we solved the issue, our Appium version was 1.21.0 and appium flutter driver version was 0.0.4, after we upgraded it to Appium 2.0 beta 40 and appium flutter driver to 1.1.0 (and above) - the build was run successfully with Flutter 3. for now we solve it in the minimum dependencies but it a great start. now we just need to configure it in our project and run it.

Thank you all for your replies!
